src/library/G36SupplyFanStatus.py

```diff
@@ -0,0 +1,62 @@
+"""
+G36 2021
+
+### Description
+
+Section 5.16.1.1,
+
+- a. Supply fan shall run when system is in the Cooldown Mode, Setup Mode, or Occupied Mode.
+- b. If there are any VAV-reheat boxes on perimeter zones, supply fan shall also run when system is in Setback Mode or Warmup Mode (i.e., all modes except unoccupied).
+
+### Verification logic
+
+```python
+if has_reheat_box_on_perimeter_zones == True:
+    if sys_mode != 'unoccupied' and supply_fan_status == 'off':
+        fail
+
+else:
+    if sys_mode in ['occupied', 'setup', 'cooldown'] and supply_fan_staus == 'off':
+        fail
+
+if ['occupied', 'unoccupied'] in sys_mode:
+  pass
+else:
+  untested
+```
+
+"""
+import pandas as pd
+from checklib import RuleCheckBase
+import numpy as np
+
+
+class G36SupplyFanStatus(RuleCheckBase):
+    points = ["sys_mode", "supply_fan_status", "has_reheat_box_on_perimeter_zones"]
+
+    def ts_verify_logic(self, t):
+        if bool(t["has_reheat_box_on_perimeter_zones"]):
+            if (t["sys_mode"].strip().lower() != "unoccupied") and (
+                not bool(t["supply_fan_status"])
+            ):
+                return False
+            return True
+        else:
+            if (
+                t["sys_mode"].strip().lower() in ["occupied", "setup", "cooldown"]
+            ) and (not bool(t["supply_fan_status"])):
+                return False
+            return True
```

Comment:
The `untested` check is missing. Fine to me but just wanted to point it out in regards to the pseudo code.

Reply:
Thanks. It is in the check_bool. The logic does not have a untested outcome at sample level. I think the way I put the pseudo code is a bit unclear. Normally the final untested scenario based on system status observation is not at time step level.